### PR TITLE
[CF-970] validate json payload

### PR DESCRIPTION
**Pull request by ChandraAddala**

### CODE REVIEW

File: app/src/main/scala/com/rackspace/prefs/PreferencesService.scala [hunk 1]
```diff
@@ -95,75 +97,93 @@ with JacksonJsonSupport {
...
+            // ';', ',', or "{}" after the top level closing curly bracelet.
+            // valid json should only have one top level valid json body enclosed by opening and closing curly bracelets.
+            if (p.nextToken != null) {
+                result = BadRequest(jsonifyError("Preferences for /" + preferenceSlug + "/" + id + " have more than one json body."))
+            }
+        }
+        catch {
+            // failed to parse jsonString, not valid
+            case e: Exception => {
+              logger.error(s"Invalid json payload:[$payload] sent for for id:[$id]. Failed parsing with exception: ${e.getMessage}")
```
Comment (ChandraAddala) [inline app/src/main/scala/com/rackspace/prefs/PreferencesService.scala]: Added a log message here.

File: app/src/main/scala/com/rackspace/prefs/PreferencesService.scala [hunk 2]
```diff
@@ -272,21 +292,43 @@ with JacksonJsonSupport {
                     }
                 }
                 catch {
-                    case e: Exception => result = BadRequest(jsonifyError(msgInvalidUrl + " Reason: " + e.getMessage()))
+                    case e: Exception => {
+                        logger.error(s"Validating container name in the url:[$containerUrl] for id:[$id] failed with exception: ${e.getMessage}")
```
Comment (ChandraAddala) [inline app/src/main/scala/com/rackspace/prefs/PreferencesService.scala]: Added another log message here. Thats it. Thats all I changed.

**Comment by ChandraAddala** [comment]
Reference PR https://github.com/rackerlabs/cloudfeeds-preferences-svc/pull/41

File: app/src/main/scala/com/rackspace/prefs/PreferencesService.scala (hunk 1 above)
Comment (shintasmith) [inline app/src/main/scala/com/rackspace/prefs/PreferencesService.scala]: Would this log the exception stack trace?

**Comment by ChandraAddala** [comment]
@shintasmith @usnavi Can you guys merge this PR. I will submit a separate PR which fixes the tests issue.

**Pull request merged**